azurerm/internal/services/healthcare/healthcare_service_resource.go

```diff
@@ -60,11 +62,27 @@ func resourceHealthcareService() *schema.Resource {
 				}, false),
 			},
 
-			"cosmosdb_throughput": {
-				Type:         schema.TypeInt,
-				Optional:     true,
-				Default:      1000,
-				ValidateFunc: validation.IntBetween(1, 10000),
+			"cosmosdb_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"throughput": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      1000,
+							ValidateFunc: validation.IntBetween(1, 10000),
+						},
+						"key_vault_key_id": {
+							Type:             schema.TypeString,
+							Optional:         true,
+							ForceNew:         true,
+							DiffSuppressFunc: diffSuppressIgnoreKeyVaultKeyVersion,
+							ValidateFunc:     keyVaultValidate.VersionlessNestedItemId,
+						},
+					},
+				},
```

Comment:
Unfortunately we can't remove properties without first deprecating them as this would be a breaking change for users with existing configurations. Can we restore the original `cosmosdb_throughput` property and rather add a toplevel `keyvault_key_id` property? We can look at adjusting the schema for v3.0 of the provider.

Reply:
I decided to use `cosmosdb_key_vault_key_id` for the attribute name, I think it better describes its purpose. If you strongly believe that `keyvault_key_id` is more suitable, I'm opt for changing it.